docs/src/main/tut/git_data.md

```diff
@@ -0,0 +1,302 @@
+
+- the repository coordinates (`owner` and `name` of the repository).
+- `message`: the new commit's message.
+- `tree`: the SHA of the tree object this commit points.
+- `parents`: the SHAs of the commits that were the parents of this commit. If omitted or empty,
+the commit will be written as a root commit. For a single parent, an array of one SHA should be provided;
+for a merge commit, an array of more than one should be provided.
+- `author`: object containing information about the author.
+
+```scala
+val createCommit = Github(accessToken)
+.gitData.createCommit(
+  "47deg",
+  "github4s",
+  "New access token",
+  "827efc6d56897b048c772eb4087f854f46256132",
+  List("d3b048c1f500ee5450e5d7b3d1921ed3e7645891"))
+
+createCommit.exec[cats.Id, HttpResponse[String]]() match {
+  case Left(e) => println("Something went wrong: s{e.getMessage}")
+  case Right(r) => println(r.result)
+}
+```
+
+The `result` on the right is the created [RefCommit][gitdata-scala].
+
+See [the API doc](https://developer.github.com/v3/git/commits/#create-a-commit) for full reference.
+
+## Blobs
+
+### Create a Blob
+
+You can create a blob using `createBlob`, it takes as arguments:
+
+- the repository coordinates (`owner` and `name` of the repository).
+- `content`: the new blob's content.
+- `encoding`: the encoding used for content. Currently, "utf-8" and "base64" are supported. Default: "utf-8".
+
+```scala
+val createBlob = Github(accessToken).gitData.createBlob("47deg", "github4s", "New access token")
+
+createBlob.exec[cats.Id, HttpResponse[String]]() match {
+  case Left(e) => println("Something went wrong: s{e.getMessage}")
+  case Right(r) => println(r.result)
+}
+```
+
+The `result` on the right is the created [RefObject][gitdata-scala].
+
+See [the API doc](https://developer.github.com/v3/git/blobs/#create-a-blob) for full reference.
+
+## Trees
+
+As you probably know, Git can be considered as a tree structure.
+Each commit creates a new node in that tree.
+Even we could assume that all the Git commands or methods provided by the API,
+are just tools that serve to navigate on this tree and to manipulate it.
+
+In the next sections, let's see how Github4s API provides some methods to wrap the Github API.
+
+### Create a Tree
+
+The tree creation API will take nested entries as well. If both a tree and a nested path modifying
+that tree are specified, it will overwrite the contents of that tree with the new path contents
+and write a new tree out.
+
+IMPORTANT: If you don't set the baseTree, the commit will be created on top of everything;
+however, it will only contain your change, the rest of your files will show up as deleted.
+
+You can create a tree using `createTree`, it takes as arguments:
+
+- the repository coordinates (`owner` and `name` of the repository).
+- `baseTree`: the SHA1 of the tree you want to update with new data.
+- `treeDataList`: list (of path, mode, type, and sha/blob) specifying a tree structure:
```

Comment:
my bad, it might be worth it to indent the list below and specify that this is a list of [TreeData][gitdata-scala]